src/engine.rs

```diff
@@ -1126,4 +1135,74 @@ mod tests {
             });
         }
     }
+
+    #[test]
+    fn test_dump_file_or_directory() {
+        let dir = tempfile::Builder::new()
+            .prefix("test_dump_file_or_directory")
+            .tempdir()
+            .unwrap();
+        let entry_data = vec![b'x'; 1024];
+
+        let mut batches = vec![vec![LogBatch::default()]];
+        let mut batch = LogBatch::default();
+        batch
+            .add_entries::<Entry>(7, &generate_entries(1, 11, Some(&entry_data)))
+            .unwrap();
+        batch.add_command(7, Command::Clean);
+        batch.put(7, b"key".to_vec(), b"value".to_vec());
+        batch.delete(7, b"key2".to_vec());
+        batches.push(vec![batch.clone()]);
+        let mut batch2 = LogBatch::default();
+        batch2.put(8, b"key3".to_vec(), b"value".to_vec());
+        batch2
+            .add_entries::<Entry>(8, &generate_entries(5, 15, Some(&entry_data)))
+            .unwrap();
+        batches.push(vec![batch, batch2]);
+
+        let cfg = Config {
+            dir: dir.path().to_str().unwrap().to_owned(),
+            ..Default::default()
+        };
+
+        let engine =
+            RaftLogEngine::open_with_file_builder(cfg, Arc::new(DefaultFileBuilder)).unwrap();
+        for bs in batches.iter_mut() {
+            for batch in bs.iter_mut() {
+                engine.write(batch, false).unwrap();
+            }
+
+            engine.sync().unwrap();
+        }
+
+        //dump dir with raft groups. 8 element in raft groups 7 and 2 elements in raft groups 8
+        let dump_it = Engine::dump(dir.path()).unwrap();
+        let total = dump_it
+            .inspect(|i| {
+                i.as_ref().unwrap();
+            })
+            .count();
+        assert!(total == 10);
+
+        //dump file
+        let file_id = FileId {
+            queue: LogQueue::Rewrite,
+            seq: 1,
+        };
+        let dump_it = Engine::dump(file_id.build_file_path(dir.path()).as_path()).unwrap();
+        let total = dump_it
+            .map(|i| {
+                i.as_ref().unwrap();
+            })
```

Comment:
```suggestion
            .map(|i| i.unwrap())
```

Reply:
if i directly follow your advice，`cargo clippy` will output the following result:

![image](https://user-images.githubusercontent.com/15794564/145935038-85930355-d007-4fc5-8a1d-a886df743c61.png)

Reply:
Oh, change it to `inspect` is enough.

Reply:
Yes,  I have changed it one place and forget to change all
```
        let total = dump_it
            .inspect(|i| {
                i.as_ref().unwrap();
            })
            .count();
        assert!(total == 10);
```

Consider the following message, `as_ref()` can't be remove

Reply:
``` 
move occurs because `*i` has type `std::result::Result<log_batch::LogItem, errors::Error>`, which does not implement the `Copy` trait
     |                 help: consider borrowing the `Result`'s content: `i.as_ref()`
```

Reply:
Yes, only map allows that.